src/features/movies/MovieList/movieListSaga.js

```diff
@@ -7,13 +7,16 @@ import {
     selectPageState,
     setMovieList,
     setGenres,
+    setLoading,
 } from "./movieListSlice";
 import { getPopularMovies } from "../../../utils/API/getPopularMovies";
 import { getGenreList } from "../../../utils/API/getGenreList";
 import { processMovieListData } from "../../../utils/API/processMovieListData";
 
 function* fetchMovieListHandler() {
     try {
+        yield put(setLoading(true));
```

Comment:
We don't need to dispatch this action:
1. I used this method in one of our previous assigments and was informed that this was a mistake. we shouldn't have reducers that only switch state this way. It's better to add the state change to another action. Moreover `loading: true` is in initialState, so we don't need this command
2. We decided to create a state property one called `status` and give it one of those values: 'loading' 'error' 'success'

Reply:
Thank you for pointing that out! 🙏 I appreciate your input. I wasn't aware that having reducers that directly switch state might not be the best practice. No one had brought this to my attention before. I will make the necessary corrections and apply the updated approach. Thanks for bringing this to my notice and sharing your knowledge! 🚀